pyparsing/core.py

```diff
@@ -1711,8 +1721,8 @@ def __call__(self, name: typing.Optional[str] = None) -> "ParserElement":
         """
         if name is not None:
             return self._setResultsName(name)
-        else:
```

Comment:
Accepted. I was under the impression that doing this kind of code in an explicit else was easier to understand then falling through to a closing return. But this looks fine to me also.